backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/service/SynchronizationService.kt

```diff
@@ -16,29 +17,18 @@ class SynchronizationService constructor(
 ) {
 
     fun synchronize(specificationId: ServiceId) {
-        val specification = specificationPersistenceService.findOne(specificationId)!!
+        val specification = specificationPersistenceService.findOne(specificationId)
 
-        val remoteAddress = specification.remoteAddress ?: ""
+        if (specification != null) {
```

Comment:
How about an "early return", i.e. first testing `specification == null` and throwing the error. Then, without an `else`, you can do the rest of the method.